edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.Kubernetes/KubernetesModuleOwner.cs

```diff
@@ -0,0 +1,31 @@
+// Copyright (c) Microsoft. All rights reserved.
+namespace Microsoft.Azure.Devices.Edge.Agent.Kubernetes
+{
+    using System;
+    using Newtonsoft.Json;
+    using Newtonsoft.Json.Serialization;
+
+    public class KubernetesModuleOwner
+    {
+        [JsonConstructor]
+        public KubernetesModuleOwner(string apiVersion, string kind, string name, string uid)
```

Comment:
Do we care if all or some are null?

Reply:
The checks are happening during KubernetesModule (the Service one) initialize. The next point this will failfast is V1ObjectMeta creation, if there were any gaps. I think this is considered an internal structure.